src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -610,11 +618,13 @@ private ProcessingBranch PrepareProcessingBranch(Block suggestedBlock, Processin
                 // we need to dig deeper to go all the way to the false (reorg boundary) head
                 // otherwise some nodes would be missing
                 bool notFoundTheBranchingPointYet = !_blockTree.IsMainChain(branchingPoint.Hash!);
-                bool notReachedTheReorgBoundary = branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0);
+                bool notReachedTheReorgBoundary = (options & ProcessingOptions.Trace) == ProcessingOptions.Trace
+                && (branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0));
```

Comment:
better naming needed, now `notReachedTheReorgBoundary` is actually `trace Or notReachedTheReorgBoundary`